common/lib/xmodule/xmodule/course_metadata_utils.py

```diff
@@ -105,12 +106,18 @@ def may_certify_for_course(certificates_display_behavior, certificates_show_befo
         certificates_show_before_end (bool): whether user can download the
             course's certificates before the course has ended.
         has_ended (bool): Whether the course has ended.
+        certificate_available_date (datetime): the date the certificate is available on for the course.
     """
     show_early = (
         certificates_display_behavior in ('early_with_info', 'early_no_info')
         or certificates_show_before_end
     )
-    return show_early or has_ended
+    past_availability_date = (
+        certificate_available_date
+        and certificate_available_date < datetime.now(utc)
+    )
+
+    return True if show_early or has_ended else past_availability_date
```

Comment:
I think this logic might still be slightly different from what we initially had in mind. As I understand it, this line is:
1. Checking if the course is set up to show certs early, and returning True if yes
2. Checking if the course has ended, and returning True if yes
3. If neither of the above, returning True if and only if the course has a certificate available date, and that date has passed.

I'm confused about the following scenario: the course has ended, but the cert available date has not passed. Does this line do what I think it does, and allow learners to view their certs even if the cert available date hasn't passed? I _think_ what we want here is to check `show_early` first, then check the cert available date, and default to `has_ended` only if the cert available date is `None`.